lib/bridge/MatrixHandler.js

```diff
@@ -129,12 +130,47 @@ MatrixHandler.prototype._handleInviteFromUser = Promise.coroutine(function*(req,
     );
 });
 
-
 // === Admin room handling ===
 MatrixHandler.prototype._onAdminMessage = Promise.coroutine(function*(req, event, adminRoom) {
     req.log.info("Received admin message from %s", event.user_id);
+
     let botUser = new MatrixUser(this.ircBridge.getAppServiceUserId());
 
+    // If an admin room has more than 2 people in it, kick the bot out
+
+    let matrixClient = this.ircBridge.getAppServiceBridge().getClientFactory().getClientAs();
+
+    let lookup = new StateLookup({
+        client : matrixClient,
+        eventTypes: ['m.room.member']
+    });
+
+    yield lookup.trackRoom(adminRoom.getId());
```

Comment:
This method will make a `/state` HTTP request the first time this is called on an object. This can be a costly operation. It is being done for each and every admin message that is received. We cannot avoid doing at least one `/state` request per room, but we can make subsequent messages faster by remembering what the previous results were and feeding in updates as and when we receive them. That is the purpose of `onEvent`, which you should use to make this a bit nicer.